core/crypto/client_tx.go

```diff
@@ -42,7 +46,11 @@ func (client *clientImpl) createDeployTx(chaincodeDeploymentSpec *obc.ChaincodeD
 	}
 
 	// Copy metadata from ChaincodeSpec
-	tx.Metadata = chaincodeDeploymentSpec.ChaincodeSpec.Metadata
+	tx.Metadata, err = getMetadata(chaincodeDeploymentSpec.GetChaincodeSpec(), tCert, attrs...)
```

Comment:
thanks for your comment @adecaro. This is a temporary solution, the correct solution implies modifying the transaction structure, we didn't want to make that change as part of the PR to not break any compatibility with existing transactions. Anyway we are not breaking retro-compatibility since this feature is disabled by default.